iterator/git.go

```diff
@@ -176,12 +179,15 @@ func (obj *Git) Validate() error {
 		return err // not that url.Parse ever really errors :/
 	}
 
-	if obj.Hash != "" && !plumbing.IsHash(obj.Hash) { // IsHash checks if len is 40
+	if obj.Hash != "" && (!plumbing.IsHash(obj.Hash) || strings.Contains(obj.Hash, separator)) { // IsHash checks if len is 40
```

Comment:
In this function you can just have the check as it's own separate if statement... No need to combine them all into a single line...

Reply:
Fixed.